core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/upgrade/DistroXUpgradeAvailabilityService.java

```diff
@@ -60,14 +60,46 @@ public boolean isRuntimeUpgradeEnabled(String userCrn) {
         }
     }
 
+    public boolean isRuntimeUpgradeEnabledByAccountId(String accountId) {
+        return entitlementService.datahubRuntimeUpgradeEnabled(accountId);
+    }
+
     public UpgradeV4Response checkForUpgrade(NameOrCrn nameOrCrn, Long workspaceId, UpgradeV4Request request, String userCrn) {
-        verifyRuntimeUpgradeEntitlement(userCrn, request);
         Stack stack = stackService.getByNameOrCrnInWorkspace(nameOrCrn, workspaceId);
         String accountId = Crn.safeFromString(userCrn).getAccountId();
         UpgradeV4Response response = stackOperations.checkForClusterUpgrade(accountId, stack, workspaceId, request);
-        List<ImageInfoV4Response> filteredCandidates = filterCandidates(accountId, stack, request, response);
-        response.setUpgradeCandidates(filteredCandidates);
-        return response;
+        UpgradeV4Response filteredResponse =
+                filterOnlyPatchUpgradesIfRuntimeUpgradeDisabled(accountId, nameOrCrn, request, response);
```

Comment:
Might it make sense to put the filterOnlyPatchUpgrade... method into the filterCandidates one?

Reply:
I copied the mechanism from the Data Lake approach.

Reply:
+1, that would be cleaner, as there are already multiple filter..... methods called from filterCandidates

Reply:
Moved it.